batched/dense/unit_test/Test_Batched_SerialPttrf.hpp

```diff
@@ -0,0 +1,401 @@
+    Kokkos::Profiling::popRegion();
+  }
+};
+
+template <typename DeviceType, typename ScalarType, typename AViewType,
+          typename BViewType, typename CViewType, typename ArgTransB>
+struct Functor_BatchedSerialGemm {
+  using execution_space = typename DeviceType::execution_space;
+  AViewType _a;
+  BViewType _b;
+  CViewType _c;
+  ScalarType _alpha, _beta;
+
+  KOKKOS_INLINE_FUNCTION
+  Functor_BatchedSerialGemm(const ScalarType alpha, const AViewType &a,
+                            const BViewType &b, const ScalarType beta,
+                            const CViewType &c)
+      : _a(a), _b(b), _c(c), _alpha(alpha), _beta(beta) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int k) const {
+    auto aa = Kokkos::subview(_a, k, Kokkos::ALL(), Kokkos::ALL());
+    auto bb = Kokkos::subview(_b, k, Kokkos::ALL(), Kokkos::ALL());
+    auto cc = Kokkos::subview(_c, k, Kokkos::ALL(), Kokkos::ALL());
+
+    KokkosBatched::SerialGemm<Trans::NoTranspose, ArgTransB,
+                              Algo::Gemm::Unblocked>::invoke(_alpha, aa, bb,
+                                                             _beta, cc);
+  }
+
+  inline void run() {
+    using value_type = typename AViewType::non_const_value_type;
+    std::string name_region("KokkosBatched::Test::SerialPttrf");
+    const std::string name_value_type = Test::value_type_name<value_type>();
+    std::string name                  = name_region + name_value_type;
+    Kokkos::RangePolicy<execution_space> policy(0, _a.extent(0));
+    Kokkos::parallel_for(name.c_str(), policy, *this);
+  }
+};
+
+template <typename DeviceType, typename ScalarType, typename LayoutType,
+          typename AlgoTagType>
+/// \brief Implementation details of batched pttrf test for random matrix
+///
+/// \param N [in] Batch size of matrix A
+/// \param BlkSize [in] Block size of matrix A
+void impl_test_batched_pttrf(const int N, const int BlkSize) {
+  using real_type      = typename real_type<ScalarType>::type;
+  using RealView2DType = Kokkos::View<real_type **, LayoutType, DeviceType>;
+  using View2DType     = Kokkos::View<ScalarType **, LayoutType, DeviceType>;
+  using View3DType     = Kokkos::View<ScalarType ***, LayoutType, DeviceType>;
+
+  View3DType A("A", N, BlkSize, BlkSize),
+      A_reconst("A_reconst", N, BlkSize, BlkSize);
+  View3DType EL("EL", N, BlkSize, BlkSize), EU("EU", N, BlkSize, BlkSize),
+      D("D", N, BlkSize, BlkSize), LD("LD", N, BlkSize, BlkSize),
+      L("L", N, BlkSize, BlkSize), I("I", N, BlkSize, BlkSize);
+  RealView2DType d("d", N, BlkSize),
+      ones("ones", N, BlkSize);  // Diagonal and sub diagnoal components
+  View2DType e_upper("e_upper", N, BlkSize - 1),
+      e_lower("e_lower", N,
+              BlkSize - 1);  // upper and lower diagnoal components
+
+  using execution_space = typename DeviceType::execution_space;
+  Kokkos::Random_XorShift64_Pool<execution_space> rand_pool(13718);
+  real_type realRandStart = 0.0, realRandEnd = 1.0;
```

Comment:
```suggestion
  real_type realRandStart, realRandEnd;
```
These get overwritten anyway so let us not confuse anyone thinking these initial values have meaning